docs/defining_new_slo.md

```diff
@@ -0,0 +1,191 @@
+
+# 1. Defining SLI
+Do a brainstorming for both developers and product owners to
+think of the important functionality and characteristics of your service
+and what matters to your customers. These will be your SLIs. 
+You should be able to represent those as a % of success or failure for the further computation.
+> Hint: Do not try to overcomplicate things right from the beginning and stop it from actually computing at least some SLIs.
+>       You can iterate on the accuracy later on.
+
+## SLI computation approaches
+There are 2 main approaches to measure the SLI
+
+### Time based
+This SLI would be based on some time interval and percentage of the time the service meets the criteria.
+Most common use-case would be an uptime computed as percentage of the time the service is operational(it might be tricky to tell what functional means).
+
+##### Advantages
+- Easy to calculate
+##### Disadvantages
+- Does not reflect if the service is used at all (overnight for example)
+- Does not reflect the impact of an outage (if only one or thousands of users were affected by an outage)
+
+### Event based
+This SLI would be computed per event of the observed phenomenon.
+Most common use-case would be observing each HTTP request result or result of some async job triggered by the user.
+
+##### Advantages
+- Actually reflects impact of the outage on users (if an outage happens but no one uses the service no one cares)
+- Reflects the amount of the outage impact (outage in prime time affects the EB more than in low traffic periods of a day)
+##### Disadvantages
+- Can be harder to compute and filter
+    - Large number of events to aggregate
+    - Some events may not be relevant to user experience (external monitoring HTTP request probes for example)
+- May be hard to alert on, in case of low frequency phenomenons (if the event occurs twice a day, single failed event might cause 50% failure ratio on short time ranges)
+- May lead to accounting repeated events of one user leading to multiplication of the impact (user refreshing the page to see if its operational).
+
+## SLI types
+These are the most common types of SLIs we encountered, of course you can think of any others that suit you but
+if possible try to stick to these common ones for consistency.
+They are based on the SRE workbook, and it makes it easier for others to understand what the SLI means.
+
+For more inspiration see https://sre.google/workbook/implementing-slos/#slis-for-different-types-of-services
+
+### Availability
+#### Time based
+Also, often referred to as an _uptime_, a ratio of the time when the service is up to the time it is down.
+You should be careful what actually means the service _is down_.
+Mostly suitable for some continuously running computations or in cases when the event based SLI is hard
+or even impossible to evaluate reliably for example SLO for relational databases(latency od success of the event depends on the user query).
```

Comment:
```suggestion
or even impossible to evaluate reliably.
```
Honestly, I would drop this example, imho it is not very relevant. We solve this by classifying events of various computation complexity to different classes (each having different SLOs). Time-based computation does  not make this problem any easier or harder opposed to request-based imho.

Reply:
This is based on real world example we experienced, but I'm ok leaving this out.